pandas/io/html.py

```diff
@@ -591,9 +591,14 @@ def _setup_build_doc(self):
     def _build_doc(self):
         from bs4 import BeautifulSoup
 
-        return BeautifulSoup(
-            self._setup_build_doc(), features="html5lib", from_encoding=self.encoding
-        )
+        bdoc = self._setup_build_doc()
+        if isinstance(bdoc, bytes) and self.encoding is not None:
```

Comment:
this is very strange

so this sometimes returns bytes? and str otherwise?

Reply:
i dont know this code particularly well, found this while tracking down `except Exception`.  AFAICT sometimes bs4 wants bytes